packages/network/src/features/pubsub/msg.rs

```diff
@@ -97,15 +94,13 @@ pub enum SourceHint {
 
 impl SourceHint {
     pub fn should_create(&self) -> bool {
-        match self {
-            SourceHint::Register { .. } => true,
-            SourceHint::Subscribe(_) => true,
-            _ => false,
-        }
+        matches!(self, SourceHint::Register { .. } | SourceHint::Subscribe(_))
     }
 }
 
 pub enum PubsubMessageError {
+    // Ask this one is never used. Please give inputs on how to use this.
+    // Did not want to simply silence the error
```

Comment:
Is there any way to read the TransportMsgHeaderError. Must we check the kind of TransportError somewhere?

Reply:
Yes, we should better Error types, maybe thiserror crate can help